docs/tutorials/thinking-in-jax.md

```diff
@@ -98,53 +101,89 @@ print(y)
 **Key concepts:**
 
 - `jax.Array` is the default array implementation in JAX.
-- The JAX array is a unified distributed datatype for representing arrays, even with physical storage spanning multiple devices
-- Automatic parallelization: You can operate over sharded `jax.Array`s without copying data onto a device using the {func}`jax.jit` transformation. You can also replicate a `jax.Array` to every device on a mesh.
+- JAX arrays may be stored on a single device, or sharded across many devices.
 
-Consider this simple example:
+When you create an array in JAX, the type is `jax.Array`:
 
 ```{code-cell}
 import jax
-from jax import Array
 import jax.numpy as jnp
 
-x = jnp.arange(5)
-isinstance(x, jax.Array)  # Returns True both inside and outside traced functions.
+x = jnp.arange(10)
+isinstance(x, jax.Array)
+```
+
+`jax.Array` is also the appropriate type annotation for functions with array inputs or outputs:
 
-def f(x: Array) -> Array:  # Type annotations are valid for traced and non-traced types.
-  return x
+```{code-cell}
+def f(x: jax.Array) -> jax.Array:
+  return jnp.sin(x) ** 2 + jnp.cos(x) ** 2
 ```
 
-The `jax.Array` type also helps make parallelism a core feature of JAX.
+JAX Array objects have a `devices` method that lets you inspect where the contents of the array are stored. In the simplest cases, this will be a single CPU device:
+
+```{code-cell}
+x.devices()
+```
+
+In general, an array may be *sharded* across multiple devices, in a manner that can be inspected via the `sharding` attribute:
+
+```{code-cell}
+x.sharding
+```
+
+In this case the sharding is on a single device, but in general a JAX array can be
+sharded across multiple devices, or even multiple hosts.
+To read more about sharded arrays and parallel computation, refer to {ref}`single-host-sharding`
 
 (thinking-in-jax-pytrees)=
-# Pytrees
+## Pytrees
 
 **Key concepts:**
 
-- JAX supports a special data structure called a pytree when you need to operate on dictionaries of lists, for example.
-- Use cases: machine learning model parameters, dataset entries, lists of lists of dictionaries.  
+- JAX supports tuples, dicts, lists, and more general containers of arrays through the
+  *pytree* abstraction.
 
-JAX has built-in support for objects that look like dictionaries (dicts) of arrays, or lists of lists of dicts, or other nested structures — they are called JAX pytrees (also known as nests, or just trees). In the context of machine learning, a pytree can contain model parameters, dataset entries, and reinforcement learning agent observations.
+Often it is convenient for applications to work with collections of arrays: for example,
+a neural network might organize its parameters in a dictionary of arrays with meaningful
+keys. Rather than handle such structures on a case-by-case basis, JAX relies on a *pytree*
+abstraction to treat such collections in a uniform matter.
```

Comment:
The key point of pytrees is that they're safe to pass to function transforms right? Like JAX will "see" the Arrays inside the pytree, but not inside arbitrary objections and those won't transform properly. Maybe worth saying something like that.

Reply:
Done